### PR TITLE
30 Projects | CRUD | Backend | Projects List

### PR DESCRIPTION
**Pull request by freddy-perez-jalafund**
Feature/30 project list

### CODE REVIEW

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjects.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+        public GetProjects(IProjectsService projectService)
+        {
+            this.projectService = projectService;
+        }
+
+        [FunctionName("projects")]
+        [OpenApiOperation(operationId: "Run", tags: new[] { "Projects" })]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(Project), Description = "Successful response")]
+        public IActionResult Run(
```
Comment (martin-callejas-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjects.cs:25]: I'm curious, why not static method/classes for the API layer? @cristian-chavez-jalafund @Raul-Gamarra-jalafund

Comment (martin-callejas-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjects.cs:25]: I mean, I don't like seeing a constructor and a field here... but I guess this is the already decided pattern, right?

Comment (freddy-perez-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjects.cs:25]: As I mentioned before, it is defined by the team in that way

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjects.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+            [HttpTrigger(AuthorizationLevel.Anonymous, "get", Route = null)] HttpRequest req)
```
Comment (rodrigo-baldiviezo-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjects.cs]: Change null to "v1/projects"

Comment (freddy-perez-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjects.cs]: Has change route

Capture

File: tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetProjectsTest.cs [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            this.getResumes = new GetProjects(this.mockService.Object);
+        }
+
+        [Fact]
+        public void GetProjects_Returns_OkObjectResult()
+        {
+            Assert.IsType<OkObjectResult>(this.getResumes.Run(this.mockRequest.Object));
+        }
+
+        [Fact]
```
Comment (Raul-Gamarra-jalafund) [inline tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetProjectsTest.cs]: we do not need all of this unit tests, please reach out to @rodrigo-baldiviezo-jalafund to check on this he has an example on testing Get All resumes API

Comment (freddy-perez-jalafund) [reply tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetProjectsTest.cs]: @Raul-Gamarra-jalafund  The names were changed in the plural and the test was modified according to what was agreed

**Review by pedro-lima-jalafund — DISMISSED** [review]

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    public class GetProject
+    {
+        private readonly IProjectService projectService;
+
+        public GetProject(IProjectService projectService)
+        {
+            this.projectService = projectService;
+        }
+
+        [FunctionName("GetProject")]
```
Comment (cristian-chavez-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs]:
```suggestion
        [FunctionName("GetProjects")]
```

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    using Jalasoft.TeamUp.Projects.Core.Interfaces;
+    using Jalasoft.TeamUp.Projects.Models;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Azure.WebJobs.Extensions.Http;
+    using Microsoft.Azure.WebJobs.Extensions.OpenApi.Core.Attributes;
+    using Microsoft.OpenApi.Models;
+
+    public class GetProject
```
Comment (cristian-chavez-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs]:
```suggestion
    public class GetProjects
```

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+        public GetProject(IProjectService projectService)
+        {
+            this.projectService = projectService;
+        }
+
+        [FunctionName("GetProject")]
+        [OpenApiOperation(operationId: "Run", tags: new[] { "Projects" })]
+        [OpenApiResponseWithBody(statusCode: HttpStatusCode.OK, contentType: "application/json", bodyType: typeof(Project[]), Description = "Successful response")]
+        [OpenApiResponseWithoutBody(statusCode: HttpStatusCode.NotFound, Description = "Resource not found")]
```
Comment (cristian-chavez-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs]: are we gonna have this scenario? or an empty array will be returned in case no projects are retrieved?

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjectById.cs [hunk 7]
```diff
@@ -13,9 +13,9 @@
 
     public class GetProjectById
     {
-        private readonly IProjectsService projectsService;
+        private readonly IProjectService projectsService;
```
Comment (cristian-chavez-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProjectById.cs]:
```suggestion
        private readonly IProjectsService projectsService;
```
please keep the names in plural

File: src/api/Jalasoft.TeamUp.Projects.API/Startup.cs [hunk 8]
```diff
@@ -18,8 +18,8 @@ public override void Configure(IFunctionsHostBuilder builder)
         {
             builder.Services.AddScoped<IHealthService, HealthService>();
             builder.Services.AddScoped<IHealthRepository, HealthRepository>();
-            builder.Services.AddScoped<IProjectsService, ProjectsService>();
-            builder.Services.AddScoped<IProjectsRepository, ProjectsRepository>();
+            builder.Services.AddScoped<IProjectService, ProjectService>();
+            builder.Services.AddScoped<IProjectRepository, ProjectRepository>();
```
Comment (cristian-chavez-jalafund) [inline src/api/Jalasoft.TeamUp.Projects.API/Startup.cs]: please could you keep the names in plural and undo these changes?

File: tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetProjectTest.cs [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+{
+    using Jalasoft.TeamUp.Projects.API.Controllers;
+    using Jalasoft.TeamUp.Projects.Core.Interfaces;
+    using Jalasoft.TeamUp.Projects.Models;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+    using Moq;
+    using Xunit;
+
+    public class GetProjectTest
```
Comment (cristian-chavez-jalafund) [inline tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetProjectTest.cs]:
```suggestion
    public class GetProjectsTest
```

File: tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetProjectTest.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    {
+        private readonly Mock<IProjectService> mockService;
+        private readonly DefaultHttpContext mockHttpContext;
+        private readonly GetProject getProject;
```
Comment (cristian-chavez-jalafund) [inline tests/api/Jalasoft.TeamUp.Projects.API.Tests/GetProjectTest.cs]:
```suggestion
        private readonly GetProjects getProjects;
```

File: src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs (hunk 6 above)
Comment (freddy-perez-jalafund) [reply src/api/Jalasoft.TeamUp.Projects.API/Controllers/GetProject.cs]: I think so because those who started the project left us those examples

**Review by alejandro-castellon — APPROVED** [review]

**Review by cristian-chavez-jalafund — APPROVED** [review]

**Review by pedro-lima-jalafund — APPROVED** [review]
good
